### PR TITLE
Bta 003 authentication

### PR DESCRIPTION
**Pull request by bkrajewski**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+
 import App from "./App";
 import * as serviceWorker from "./serviceWorker";
-import { BrowserRouter } from "react-router-dom";
 
 ReactDOM.render(
-  <BrowserRouter>
-    <App />
-  </BrowserRouter>,
+  <Provider store={createStore(reducers)}>
```
Comment (paweljedrzejczyk) [inline src/index.js]: I think `createStore` should be called outside of ReactDOM.render,
```js
const store = creatStore(reducers);
(...)
ReactDOM.render(
  <Provider store={store}>
```

File: src/components/ui-components/Button/Button.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    padding: ${({ theme }) => `${theme.spacingTiny} ${theme.spacingLarge}`};
+    font-size: ${({ theme }) => theme.fontSizeNormal};
+    border: ${({isPrimary, theme}) => isPrimary ? `3px solid ${theme.colors.japaneseLaurelDark}` : `3px solid ${theme.colors.trout}`};
+    border-radius: 5px;
+    background-color: ${({isPrimary, theme}) => isPrimary ? theme.colors.japaneseLaurel: theme.colors.white};
+    color: ${({isPrimary, theme}) => isPrimary ? theme.colors.white : theme.colors.trout};
+    margin-left: ${({ theme }) => theme.spacingNormal}; 
+    margin-right: ${({ theme }) => theme.spacingNormal}; 
+    opacity: ${({disabled}) => disabled ? '0.7' : '1'};
+    cursor: ${({disabled}) => disabled || 'pointer'};
```
[the remark this answers is not in the document]
Comment (bkrajewski) [reply src/components/ui-components/Button/Button.js]: It works like:
    if(!disabled) { cursor: pointer }
or 
    !disabled && cursor: pointer

It works fine in the project, do you think it should stay this way?

File: src/App.js [hunk 3]
```diff
@@ -27,47 +33,88 @@ const GlobalStyle = createGlobalStyle`
...
 `;
 
-const App = () => {
-  const [showMobileSidebar, toggleMobileSidebar] = useState((false));
-  const [showDesktopSidebar, toggleDesktopSidebar] = useState(true);
+const App = props => {
+  const showMobileSidebar = useToggleValue(false);
+  const showDesktopSidebar = useToggleValue(true);
   const [availableSidebar] = useState(() => {
     if (window.innerWidth <= theme.desktopWidth) {
```
[the remark this answers is not in the document]
Comment (bkrajewski) [reply src/App.js]: Im aware of it, but i'd like to leave it as it is for now. I think sizeme may not be used in this project in any other case than to decide which sidebar should be displayed (mobile or desktop). I didn't decide yet if i want support this kind of screen resolution jumps. What do you think about it?

File: src/index.js (hunk 1 above)
Comment (bkrajewski) [reply src/index.js]: Thanks, great tip! Could you tell me why it should be done this way? Is it due to creating this function anew every time index.js rerenders?

File: src/components/ui-components/Button/Button.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (paweljedrzejczyk) [reply src/components/ui-components/Button/Button.js]: I was not aware of this behavior, if that's the case :+1:

File: src/index.js (hunk 1 above)
Comment (paweljedrzejczyk) [reply src/index.js]: Yes, theoretically ReactDOM.render can be called multiple times, better be safe and create the store only once.

File: src/App.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (paweljedrzejczyk) [reply src/App.js]: Ideally your app should only rely on props and state. 
`window.innerWidth` is external value that can change in the meantime and if you use that value you should respond to its change.

In real world apps you should use `react-sizeme` or something similar, even created by yourself, just to learn how it works ;)

File: src/App.js [hunk 4]
```diff
@@ -41,29 +41,37 @@ const PageContent = styled.div`
...
     } else {
       return "desktop";
     }
   });
 
+  useEffect(() => {
+    const screenSize =
+      props.size.width <= theme.desktopWidth ? "mobile" : "desktop";
+    if (screenSize !== displayMode) {
+      setDisplayMode(screenSize);
```
Comment (paweljedrzejczyk) [inline src/App.js]: setting state from props is an anti-pattern, you should use props as the source of truth when using react-sizeme

with
```js
withSize(size => ({
  isMobile: size.width <= theme.desktopWidth,
  isDesktop: size.width > theme.desktopWidth,
}))(App)
```

you will have access to `isMobile` and `isDesktop` props. You can also do it with:
```js
withSize(size => ({
  displayMode: props.size.width <= theme.desktopWidth ? "mobile" : "desktop"
}))(App)
```

**Comment by bkrajewski** [comment]
@paweljedrzejczyk Please check app.js lines 82-85

**Comment by paweljedrzejczyk** [comment]
> @paweljedrzejczyk Please check app.js lines 82-85

Most probably because of the order of HOCs. Try putting `react-sizeme` before `connect`. Also if you have more than 1 or 2 HOC it's better to use `compose` function for better readability. You could do:

```js

import { compose } from 'redux';

export default compose(
  connect(
    mapStateToProps,
    { signIn, signOut }
  ),
  withSize(),
  withRouter,
)(App);
```

the compose function "accepts arguments" in reversed order, so withRouter is last here.

**Comment by bkrajewski** [comment]
> > @paweljedrzejczyk Please check app.js lines 82-85
> 
> Most probably because of the order of HOCs. Try putting `react-sizeme` before `connect`. Also if you have more than 1 or 2 HOC it's better to use `compose` function for better readability. You could do:
> 
> ```js
> import { compose } from 'redux';
> 
> export default compose(
>   connect(
>     mapStateToProps,
>     { signIn, signOut }
>   ),
>   withSize(),
>   withRouter,
> )(App);
> ```
> 
> the compose function "accepts arguments" in reversed order, so withRouter is last here.

I tried all possible configurations, tried to use sizeMe() hoc instead of withSize() - none of them working. Any other ideas?

**Pull request merged**